scrapy/utils/reqser.py

```diff
@@ -20,6 +19,7 @@ def request_to_dict(request, spider=None):
     if callable(eb):
         eb = _find_method(spider, eb)
     d = {
+        '__class__': request.__class__,
```

Comment:
This doesn't work well with `scrapy.squeues.MarshalFifoDiskQueue` and `scrapy.squeues.MarshalLifoDiskQueue`, the resulting dicts cannot be serialized. I'm working on it.